tests/functional/common/functions.sh

```diff
@@ -345,4 +345,15 @@ count() {
 
 trap onError ERR
 
+requiresUnprivilegedUserNamespaces() {
+  if [[ -f /proc/sys/kernel/apparmor_restrict_unprivileged_userns ]] && [[ $(< /proc/sys/kernel/apparmor_restrict_unprivileged_userns) -eq 1 ]]; then
```

Comment:
This test should also be added to `userNamespacesSupported()` in `src/libutil/linux/namespaces.cc`. In fact, if we check for it in Nix itself, then maybe the tests will succeed even if unprivileged user namespaces are restricted.